src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabServerPullRequestDecorator.java

```diff
@@ -87,29 +96,31 @@ public GitlabServerPullRequestDecorator(Server server, ScmInfoRepository scmInfo
     }
 
     @Override
-    public void decorateQualityGateStatus(AnalysisDetails analysis, UnifyConfiguration unifyConfiguration) {
+    public void decorateQualityGateStatus(AnalysisDetails analysis, AlmSettingDto almSettingDto,
+                                          ProjectAlmSettingDto projectAlmSettingDto) {
         LOGGER.info("starting to analyze with " + analysis.toString());
         String revision = analysis.getCommitSha();
 
         try {
-            final String hostURL = unifyConfiguration.getRequiredServerProperty(PULLREQUEST_GITLAB_URL);
-            final String apiToken = unifyConfiguration.getRequiredServerProperty(PULLREQUEST_GITLAB_TOKEN);
-            final String repositorySlug = unifyConfiguration.getRequiredProperty(PULLREQUEST_GITLAB_REPOSITORY_SLUG);
+            final String apiURL = Optional.ofNullable(StringUtils.stripToNull(almSettingDto.getUrl()))
+                .orElse(analysis.getScannerProperty(PULLREQUEST_GITLAB_INSTANCE_URL)
+                .orElseThrow(() -> new IllegalStateException(String.format(
+                            "Could not decorate Gitlab merge request. '%s' has not been set in scanner properties",
+                            PULLREQUEST_GITLAB_INSTANCE_URL))));
+            final String apiToken = almSettingDto.getPersonalAccessToken();
+            final String projectId = analysis.getScannerProperty(PULLREQUEST_GITLAB_PROJECT_ID).orElseThrow(
+                    () -> new IllegalStateException(String.format(
+                            "Could not decorate Gitlab merge request. '%s' has not been set in scanner properties",
+                            PULLREQUEST_GITLAB_PROJECT_ID)));
             final String pullRequestId = analysis.getBranchName();
 
-            final boolean summaryCommentEnabled = Boolean.parseBoolean(unifyConfiguration.getRequiredServerProperty(PULL_REQUEST_COMMENT_SUMMARY_ENABLED));
-            final boolean fileCommentEnabled = Boolean.parseBoolean(unifyConfiguration.getRequiredServerProperty(PULL_REQUEST_FILE_COMMENT_ENABLED));
-            final boolean deleteCommentsEnabled = Boolean.parseBoolean(unifyConfiguration.getRequiredServerProperty(PULL_REQUEST_DELETE_COMMENTS_ENABLED));
-
-            final String restURL = String.format("%s/api/v4", hostURL);
-            final String userURL = restURL + "/user";
-            final String projectURL = restURL + String.format("/projects/%s", URLEncoder.encode(repositorySlug, StandardCharsets.UTF_8.name()));
+            final String projectURL = apiURL + String.format("/projects/%s", URLEncoder.encode(projectId, StandardCharsets.UTF_8.name()));
```

Comment:
I'm trying to use Sonar 8.3 with selft Hosted Gitlab 12, and at this line, URLEncoder.encode is transforming the slashes of my project slug('/group/subgroup/project') in '%2F', and, of course, Gitlab is returning "HTTP/1.1 404 Not Found".

Reply:
I got It wrong. As in [gitlab-api-docs](https://docs.gitlab.com/ee/api/README.html#namespaced-path-encoding), I could check this line is right and I put gitlab-url at "ALM Integrations" without /api/v4.